### ISSUE TITLE
[PATCH] Logic error in jabber.client

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10358]
|<img alt="Maniac's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Maniac reported|
|-|-|
|Trac ID|trac#434|
|Type|defect|
|Created|2003-12-15 04:30:57Z|

Attachments:

* [client.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/894/8949eb0b6a2ef0595f9ef639e167d6209c3ccc60/918bab7a34a82ccdc56504d97a38bacd3f9551ca.patch) (680 bytes) - added by Maniac on 2003-12-15 04:30:57Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__434 434
type__defect defect
reporter__Maniac Maniac
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1071462657000000 1071462657000000
changetime__1469704406887656 1469704406887656
version__ 
owner__dizzyd dizzyd
cc__dizzyd cc__Maniac
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Maniac's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Maniac commented|
|-|-|

```
#!html
<pre>
If an incorrect password is provided AUTH_FAILED_EVENT is never called. 

This patch addresses this
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="dizzyd's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|dizzyd commented|
|-|-|

```
#!html
<pre>
Fixed -- was dispatching wrong event on auth result error
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub commented|
|-|-|

In changeset ef1b5509e10d52e2019cd577b44d2101d8d974d1
```
#!CommitTicketReference repository="" revision="ef1b5509e10d52e2019cd577b44d2101d8d974d1"
Merge pull request #434 from twisted/8698-rodrigc-conch-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8698

Port twisted.conch.test.test_insults to Python 3
```
